pdns/pdns_recursor.cc

```diff
@@ -1773,7 +1798,7 @@ static void startDoResolve(void *p)
     dq.extendedErrorExtra = &dc->d_extendedErrorExtra;
     dq.meta = std::move(dc->d_meta);
 
-    if(ednsExtRCode != 0) {
+    if(ednsExtRCode != 0 || dc->d_mdp.d_header.opcode == Opcode::Notify) {
```

Comment:
This feels a bit late. We've been allocating and initializing a bunch of things for nothing

Reply:
yes, but doing it earlier will make the compiler complain about skipping of initializations of vars used later

Reply:
we need to kill the goto

Reply:
yes, this whole function needs a rewrite... but not as part of this PR